### PR TITLE
Snapshot create across the namespace

### LINKED ISSUE #1610: Snapshot creation failed when the volume is created in the non default namespace.
BUG REPORT

- Created volume in non default namespace and tried to created kubectl snapshot..
-The snapshot creation failed.

maya-apiserver looking for default namespace during snapshot creation rather than the specified volume namespace.

```
2018/06/13 13:33:49.374839 [DEBUG] http: Request /latest/snapshots/create/ (POST)
I0613 13:33:49.375036       7 snapshot_endpoint.go:43] Processing Volume create snapshot request
I0613 13:33:49.380997       7 snapshot_endpoint.go:64] Processing snapshot-create request of volume: pvc-ead44a0a-6f05-11e8-9bb5-02b983f0a4db
I0613 13:33:49.381437       7 volume_endpoint.go:137] Processing Volume read request

I0613 13:33:49.425233       7 k8s.go:482] Fetching info on volume 'default: pvc-ead44a0a-6f05-11e8-9bb5-02b983f0a4db'
W0613 13:33:49.491508       7 k8s.go:503] Missing controller Deployment(s) for volume 'default: pvc-ead44a0a-6f05-11e8-9bb5-02b983f0a4db'
W0613 13:33:49.555613       7 k8s.go:523] Missing Replica Deployment(s) for volume 'default: pvc-ead44a0a-6f05-11e8-9bb5-02b983f0a4db'
W0613 13:33:49.602174       7 k8s.go:540] Missing Controller Pod(s) for volume 'default: pvc-ead44a0a-6f05-11e8-9bb5-02b983f0a4db'
W0613 13:33:49.732810       7 k8s.go:557] Missing Replica Pod(s) for volume 'default: pvc-ead44a0a-6f05-11e8-9bb5-02b983f0a4db'
W0613 13:33:49.782611       7 k8s.go:573] Missing Controller Service(s) for volume 'default: pvc-ead44a0a-6f05-11e8-9bb5-02b983f0a4db'
2018/06/13 13:33:49.782776 [ERR] http: Request POST /latest/snapshots/create/, error: Volume 'pvc-ead44a0a-6f05-11e8-9bb5-02b983f0a4db' not found
2018/06/13 13:33:49.782857 [DEBUG] http: Request /latest/snapshots/create/ (408.052288ms)
```

### PR DESCRIPTION
**Pull request by prateekpandey14**
This commit will allow snapshot-create across app namespace.
fixes: https://github.com/openebs/openebs/issues/1610
Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

### CODE REVIEW

File: openebs/pkg/v1/snapshot_api.go [hunk 1]
```diff
@@ -87,7 +88,7 @@ func (v OpenEBSVolume) CreateSnapshot(volName string, snapName string) (string,
 }
 
 // ListVolume to get the info of Vsm through a API call to m-apiserver
-func (v OpenEBSVolume) ListSnapshot(volName string, snapname string, obj interface{}) error {
+func (v OpenEBSVolume) ListSnapshot(volName string, snapname string, namespace string, obj interface{}) error {
```
Comment (kmova) [inline openebs/pkg/v1/snapshot_api.go:91]: Is this function currently not used?

File: snapshot/pkg/volume/openebs/processor.go [hunk 2]
```diff
@@ -233,7 +233,7 @@ func (h *openEBSPlugin) SnapshotRestore(snapshotData *crdv1.VolumeSnapshotData,
 	volSize := pvc.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]
 	volumeSpec.Metadata.Labels.Storage = volSize.String()
 	volumeSpec.Metadata.Labels.Namespace = pvc.Namespace
-	volumeSpec.Metadata.Labels.PersistentVolumeClaim = options.PVC.ObjectMeta.Name
+	volumeSpec.Metadata.Labels.PersistentVolumeClaim = pvc.ObjectMeta.Name
```
Comment (kmova) [inline snapshot/pkg/volume/openebs/processor.go:236]: good catch!

**Review by kmova — APPROVED** [review]
LGTM

File: openebs/pkg/v1/snapshot_api.go (hunk 1 above)
Comment (prateekpandey14) [reply openebs/pkg/v1/snapshot_api.go:91]: Yes, snapshot list is just for the logging purpose , not really needed by snapshot-controller.

**Pull request merged**